app/src/main/java/team7/voluntime/Fragments/Common/UserProfileFragment.java

```diff
@@ -1,42 +1,148 @@
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
 
 import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
 
+import butterknife.BindView;
 import butterknife.ButterKnife;
 import butterknife.OnClick;
-import team7.voluntime.Activities.LoginActivity;
 import team7.voluntime.Activities.MainActivity;
 import team7.voluntime.R;
 
+import static android.view.View.VISIBLE;
+
 /**
  * A simple {@link Fragment} subclass.
  */
 public class UserProfileFragment extends Fragment {
 
-    private final static String TAG = "MapFragment";
+    private final static String TAG = "UserProfileFragment";
+
+    //Local variables used in the store database values
+    private String Type;
+    private String UserName;
+    private String DOB;
+    private String Phone;
+    private String Address;
+    @BindView(R.id.charityStuff)
+    LinearLayout charityStuff;
+    @BindView(R.id.uCat)
+    TextView uCat;
+    @BindView(R.id.uDesc)
+    TextView uDesc;
+    private String GEN;
+    private String Category;
+    private String Desc;
+
+    //Bindings
+    @BindView(R.id.edit)
+    ImageView editLogo;
+
+    @BindView(R.id.volunteerStuff)
+    LinearLayout volunteerStuff;
+    //Local database refrence
+    private FirebaseAuth mAuth;
+
+    @BindView(R.id.uName)
+    TextView uName;
+    private FirebaseAuth.AuthStateListener mAuthStateListener;
+    private FirebaseUser mUser;
+
+    @BindView(R.id.uType)
+    TextView uType;
+
+    @BindView(R.id.uEmail)
+    TextView uEmail;
+
+    @BindView(R.id.uPhone)
+    TextView uPhone;
+
+    @BindView(R.id.uAddress)
+    TextView uAddress;
+
+    @BindView(R.id.uGen)
+    TextView uGen;
+
+    @BindView(R.id.uDob)
+    TextView uDob;
 
 
     public UserProfileFragment() {
         // Required empty public constructor
     }
 
-
     @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+
         // Inflate the layout for this fragment
         View v = inflater.inflate(R.layout.fragment_user_profile, container, false);
         ButterKnife.bind(this, v);
+
+        //Get Current logedin instance from database
+        mAuth = FirebaseAuth.getInstance();
+        mUser = mAuth.getCurrentUser();
+        //Get Accoount type and pass it into getType() method to return corects string path
+        DatabaseReference typeRef = FirebaseDatabase.getInstance().getReference(getType());
+
+        typeRef.child(mUser.getUid()).addValueEventListener(new ValueEventListener() {
```

Comment:
You could add another child ("Profile") here, and then when you get the value of each element you wouldnt need to have Profile all those times. The only reason you need it now is for "AccountType" but because we already know what the account type will be cause you're checking for that exact string, so we dont need to get the value of that. (If this doesnt make sense let me know)